app/assets/javascripts/admin/models/theme.js.es6

```diff
@@ -269,6 +269,16 @@ const Theme = RestModel.extend({
     return this.saveChanges("child_theme_ids");
   },
 
+  addParentTheme(theme) {
+    let parentThemes = this.parentThemes;
+    if (!parentThemes) {
+      parentThemes = [];
+      this.set("parentThemes", parentThemes);
+    }
+    parentThemes.removeObject(theme);
```

Comment:
This is fishy - why are you removing and then adding the same thing? Is it to avoid duplicates? In which case `addObject` will do that.